### ISSUE TITLE
Brave Nightly crashing if #native-brave-wallet is set to Disabled and creating new profile

### ISSUE THREAD

**Issue by Saoiray** [brave/brave-browser#30466]
## Description 
Brave Nightly crashes when attempting to create a new profile if you have Enable Brave Wallet disabled. 

## Steps to Reproduce

1. Go to `brave://flags` and set Enable Brave Wallet to **Disabled**
2. Attempt to create a new profile

## Actual result:
Brave crashes. 
Two crash reports from different users:
* ae720100-8e2d-1b0a-0000-000000000000
* c95a0300-8e2d-1b0a-0000-000000000000

## Expected result:
New profile is created

## Reproduces how often: 
Every time you try to create a new profile. 

## Brave version (brave://version info)
Originally reported on 1.53.47
Verified by me on 1.53.50
And just tested once more on newly updated 1.53.54

## Version/Channel Information:

- Can you reproduce this issue with the current release? **No**
- Can you reproduce this issue with the beta channel?  **Did not test**
- Can you reproduce this issue with the nightly channel? **Yes**

## Other Additional Information:

Topic open on Brave Community at https://community.brave.com/t/crash-when-attempting-to-show-user-profile-picker-with-native-brave-wallet-flag-disabled/487734

**Comment by iefremov** [comment]
probably something changed upstream, b/c i can't repro it on older versions. 
```
[ 00 ] EthereumRemoteClientService::IsCryptoWalletsReady() const ( __tree:2487 )
[ 01 ] BraveContentBrowserClient::HandleURLOverrideRewrite(GURL*, content::BrowserContext*) ( brave_content_browser_client.cc:1022 )
[ 02 ] content::BrowserURLHandlerImpl::RewriteURLIfNecessary(GURL*, content::BrowserContext*, bool*) ( browser_url_handler_impl.cc:155 )
[ 03 ] content::NavigationControllerImpl::CreateNavigationEntry(GURL const&, content::Referrer, absl::optional<url::Origin>, absl::optional<GURL>, content::SiteInstance*, ui::PageTransition, bool, std::Cr::basic_string<char, std::Cr::char_traits<char>, std::Cr::allocator<char>> const&, content::BrowserContext*, scoped_refptr<network::SharedURLLoaderFactory>, bool) ( navigation_controller_impl.cc:421 )
[ 04 ] content::NavigationControllerImpl::LoadURLWithParams(content::NavigationController::LoadURLParams const&) ( navigation_controller_impl.cc:3730 )
[ 05 ] content::NavigationControllerImpl::LoadURL(GURL const&, content::Referrer const&, ui::PageTransition, std::Cr::basic_string<char, std::Cr::char_traits<char>, std::Cr::allocator<char>> const&) ( navigation_controller_impl.cc:1267 )
[ 06 ] ProfilePickerView::ShowScreen(content::WebContents*, GURL const&, base::OnceCallback<void ()>) ( profile_picker_view.cc:399 )
[ 07 ] non-virtual thunk to ProfilePickerView::ShowScreenInPickerContents(GURL const&, base::OnceCallback<void ()>) ( profile_picker_view.cc:428 )
[ 08 ] ProfileManagementFlowController::SwitchToStep(ProfileManagementFlowController::Step, bool, base::OnceCallback<void (bool)>, base::OnceCallback<void ()>) ( profile_management_flow_controller.cc:38 )
[ 09 ] ProfilePickerFlowController::Init(base::OnceCallback<void (bool)>) ( profile_picker_flow_controller.cc:266 )
[ 10 ] ProfilePickerView::Init(Profile*) ( profile_picker_view.cc:629 )
[ 11 ] ProfilePickerView::OnPickerProfileCreated(Profile*) ( profile_picker_view.cc:588 )
[ 12 ] ProfileManager::OnProfileCreationFinished(Profile*, Profile::CreateMode, bool, bool) ( callback.h:152 )
[ 13 ] ProfileImpl::OnPrefsLoaded(Profile::CreateMode, bool) ( profile_impl.cc:846 )
[ 14 ] PrefNotifierImpl::OnInitializationCompleted(bool) ( callback.h:152 )
[ 15 ] policy::ConfigurationPolicyPrefStore::OnPolicyServiceInitialized(policy::PolicyDomain) ( configuration_policy_pref_store.cc:113 )
[ 16 ] policy::PolicyServiceImpl::MergeAndTriggerUpdates() ( policy_service_impl.cc:484 )
[ 17 ] non-virtual thunk to base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() ( callback.h:152 )
[ 18 ] base::MessagePumpGlib::Run(base::MessagePump::Delegate*) ( message_pump_glib.cc:670 )
[ 19 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) ( thread_controller_with_message_pump_impl.cc:651 )
[ 20 ] base::RunLoop::Run(base::Location const&) ( thread_controller_with_message_pump_impl.cc:0 )
[ 21 ] content::BrowserMainLoop::RunMainMessageLoop() ( browser_main_loop.cc:1072 )
[ 22 ] content::BrowserMain(content::MainFunctionParams) ( browser_main_runner_impl.cc:159 )
[ 23 ] content::ContentMainRunnerImpl::Run() ( content_main_runner_impl.cc:706 )
[ 24 ] content::ContentMain(content::ContentMainParams) ( content_main.cc:326 )
[ 25 ] ChromeMain ( chrome_main.cc:187 )
```

**Comment by xBZZZZ** [comment]
oldest broken nightly is [v1.53.43](https://github.com/brave/brave-browser/releases/tag/v1.53.43)
[v1.53.42](https://github.com/brave/brave-browser/releases/tag/v1.53.43) works fine

**Comment by xBZZZZ** [comment]
I think this is fixed because no crash [v1.58.3](https://github.com/brave/brave-browser/releases/tag/v1.58.3)
@Saoiray close this issue
